includes/Core/Authentication/Authentication.php

```diff
@@ -633,7 +633,8 @@ private function inline_js_setup_data( $data ) {
 		$data['isAuthenticated']    = ! empty( $access_token );
 		$data['requiredScopes']     = $auth_client->get_required_scopes();
 		$data['grantedScopes']      = ! empty( $access_token ) ? $auth_client->get_granted_scopes() : array();
-		$data['needReauthenticate'] = $data['isAuthenticated'] && $auth_client->needs_reauthentication();
+		$data['unsatisfiedScopes']  = ! empty( $access_token ) ? $auth_client->get_unsatisfied_scopes() : array();
+		$data['needReauthenticate'] = $auth_client->needs_reauthentication();
```

Comment:
`needs_reauthentication` already checks for the access token internally, so no need to double-check it here.